### DISCUSSION TITLE
Schedule functionality, or some function to generate files

### DISCUSSION THREAD

**Discussion by Arv18** [forcedotcom/dataloader#1219] [Ideas]
Hi

I think we could add a schedule functionality, or some function to generate files/configurations that can be used on the command line for a scheduled import or export.

This [Salesforce documentation](https://developer.salesforce.com/docs/atlas.en-us.dataLoader.meta/dataLoader/using_the_command_line_interface.htm) allows you to do it in steps that require code in the Windows command line. The goal would be to do something similar but automated from a user-friendly interface.

More comments, ideas, etc. are welcome.

**Comment by jwesterman-salesforce** [comment]
Hi,
Thanks for the suggestion but scheduling capabilities are out of scope for Data Loader.  If you wish schedule the execution of Data Loader, then it can be done from the OS (as you describe) or a scheduling tool running a batch file.
Thanks,
Jeremy

**Comment by ashitsalesforce** [comment]
Hi @Arv18 , to Jeremy's point, a number of utilities exist on Windows and Mac to set up scheduled execution of Data Loader in CLI mode.

For example, you can schedule Data Loader to execute in CLI mode at specific interval or when a certain event occurs using Windows Task Manager as outlined [here](https://www.windowscentral.com/how-create-automated-task-using-task-scheduler-windows-10). While the steps outlined are for Windows 10, they also work in Windows 11. 

Similar utilities exist on Mac. [Here](https://apple.stackexchange.com/questions/426040/run-shell-scripts-on-a-schedule) is a discussion regarding such utilities that you may find useful.

**Comment by Arv18** [comment]
Hi friends!

Thanks for the answers and I understand the suggestions, but the main objective of the idea is to make these steps in the OS (mainly the coding ones) **automated from a friendlier interface only for SF**, such as Jitterbit SF Data Loader (where you select a file to import, a schedule and some extra options for the log, etc), or another independent java program (which also allows scheduling an execution in the backend) for salesforce. 

If perhaps it is not the objective of data loader to implement a friendly tool using its files generated by the CLI for scheduled execution, I would start to remember that in these cases it could be an app tool only for this objective or others that are not in scope, something like a Data Loader App Tools or Extended or Out Scope... :)

I hope we continue to have more ideas and comments to determine the best solutions for anyone who needs it. Thank you!

**Comment by ashitsalesforce** [comment]
Hi @Arv18 , [dataloader.io ](https://dataloader.io/) is perhaps the right Salesforce offering for what you are looking for.

**Reply by Arv18** [reply]
Thanks @ashitsalesforce , I will consider it if the idea expands into a web application. 👍🏾
